express-post-json/server.ts

```diff
@@ -0,0 +1,32 @@
+import express from 'express';
+
+type Grade = { id: number; name: string; course: string; score: number };
+type Grades = Record<number, Grade>;
+
+let id = 1;
+
+const grades: Grades = {};
+
+const app = express();
+
+app.get('/api/grades', (req, res) => {
+  const gradeArr = [];
+  for (const key in grades) {
+    gradeArr.push(grades[key]);
+  }
+  res.json(gradeArr);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  req.body.id = id;
+  const newGrade = req.body;
```

Comment:
You'd just as well declare this to be of type `Grade`. That will help catch errors further on in the function.